### DISCUSSION TITLE
Which Password Manager we should move to for magma?

### DISCUSSION THREAD

**Discussion by ShubhamTatvamasi** [magma/magma#6276] [General]
We have to move from Facebook's Lastpass to some new Password Manager. Any suggestions?

**Comment by ShubhamTatvamasi** [comment]
Some questions:

1. Should we look into Open Source?
2. Is it okay to self-host? e.g. Vault by HashiCorp

**Reply by ShubhamTatvamasi** [reply]
If we can self-host, we can also consider [passbolt](https://www.passbolt.com/).

**Reply by ShubhamTatvamasi** [reply]
I personally like [bitwarden](https://bitwarden.com), because it's open source and a lot cheaper than lastpass.

**Reply by artblee** [reply]
credstash also works

**Reply by artblee** [reply]
https://github.com/fugue/credstash

**Comment by hcgatewood** [comment]
From a previous discussion: I think the solution here is (a) use [SOPS](https://github.com/mozilla/sops) for *as much as possible*, and then use an open-source solution for the top-level keys

Ref: see the end of https://github.com/magma/magma/discussions/5587

**Reply by artblee** [reply]
SOPS present dependancies and you still just endup putting encrypted secrets in the scm repo which break compliance.

**Reply by hcgatewood** [reply]
I don't understand either of these critiques

1. How is `brew install sops` more of a dependency than installing a separate password manager? Or, for that matter, running our own pw manager deployment?
2. If I'm missing something please explain, but I fail to understand how storing encrypted secrets in a repo is less compliant (is there a particular standard you're referencing?) than storing them e.g. in our CircleCI env. Seems to me the difficulty in exfiltrating them is ~equivalent. I get that there may be other, non-CI secrets we want to store, but at minimum I think storing CI secrets via SOPs is, at minimum, *no worse* than the existing and presented options

**Comment by artblee** [comment]
https://secrethub.io/ is another one

**Reply by artblee** [reply]
they have an aliance with 1pass

**Reply by ShubhamTatvamasi** [reply]
This is again very costly.
